WMDE_FR_2025_ipad_DE_01

Sperling-0 opened the pull request:
Both banners

    The banners are based on the [control banner](https://de.wikipedia.org/?banner=B24_WMDE_ipad_DE_01_ctrl&devMode) of the 2024 setup.
    Both banners
        use the progress bar.
        use the campaign day sentence.
        use the visitors-vs-donors sentence.
            have the header "Wikipedia ist unverkäuflich" on each slide instead of "Ist Ihnen Wikipedia 5 € wert?"
            use the imperative headline "Bitte spenden Sie 5 € jährlich!" in the upsell.
            have the new use of funds page.
            have a "Habe schon gespendet" link (maybe at the top left within the red box?).
            have the same slide interval as mobile.
            have 22,89 € as the average donation in the copy.
            support the dark mode
        do not have the soft close.

Variant banner

    has the 25 € amount button highlighted (See the [WMF Banner](https://en.wikipedia.org/wiki/NASA?banner=B2526_0701_en6C_dsk_p1_lg_template&country=US))
    We will use the styles from the amount cheering in the de desktop banner for this
    if possible: the highlighted amount button disappears if I click on monthly or yearly in the banner form.
    If there isn't enough room for the element the height of the fields can be reduced a little
    
Ticket: https://phabricator.wikimedia.org/T405892

Sperling-0 committed:
Prepare for campaign WMDE_FR_2025_ipad_DE_01

Ticket: https://phabricator.wikimedia.org/T405892

Sperling-0 committed:
Implement CTRL for WMDE_FR_2025_ipad_DE_01

- Use the progress bar
- Use the campaign day sentence
- Use the visitors-vs-donors sentence
- Have the header "Wikipedia ist unverkäuflich" on each slide instead of "Ist Ihnen Wikipedia 5 € wert?"
- Use the imperative headline "Bitte spenden Sie 5 € jährlich!" in the upsell
- Have the new use of funds page
- Have a "Habe schon gespendet" link
- Have the same slide interval as mobile
- Have 22,89 € as the average donation in the copy
- Do not have the soft close
- Support the dark mode

Ticket: https://phabricator.wikimedia.org/T405892

Sperling-0 committed:
Prepare VAR for WMDE_FR_2025_ipad_DE_01

Ticket: https://phabricator.wikimedia.org/T405892

Sperling-0 committed:
Implement VAR for WMDE_FR_2025_ipad_DE_01

Ticket: https://phabricator.wikimedia.org/T405892

Sperling-0 committed:
Use FooterAlreadyDonated component

- Remove "Habe schon gespendet" link at the top left (within the red box) which was added in the previous commit

Ticket: https://phabricator.wikimedia.org/T405892

Abban commented on banners/pad/WMDE_FR_2025_ipad_DE_01/styles/styles_var.scss:
```diff
@@ -0,0 +1,50 @@
+@use '@src/themes/Treedip/Footer/FooterAlreadyDonated';
+@use '@src/themes/Treedip/Footer/BannerFooter' with (
+	$right-column-width: 35%
+);
```
```suggestion
@use '@src/themes/Treedip/Footer/FooterAlreadyDonated';
```

Abban replied:
The right column should be 300px, but that's the default so you don't need to pass it as a parameter

Abban commented on banners/pad/WMDE_FR_2025_ipad_DE_01/styles/styles.scss:
```diff
@@ -0,0 +1,50 @@
+@use '@src/themes/Treedip/Footer/FooterAlreadyDonated';
+@use '@src/themes/Treedip/Footer/BannerFooter' with (
+	$right-column-width: 35%
+);
```
```suggestion
@use '@src/themes/Treedip/Footer/FooterAlreadyDonated';
```

Abban reviewed (changes requested):
Larissa and I had a chat about the popup and we both agreed that it should use the blue colour that the desktop uses. You can also remove the shadow.

<img width="398" height="61" alt="Image" src="https://github.com/user-attachments/assets/cbbe3d77-c818-44a3-854c-2436d0e78866" />

Sperling-0 committed:
Update the sticky notes with new text and color

- The sticky note is hidden when the error message is shown and displayed again when the user resolved the error
- Remove redundant Footer styles
- Update content

Ticket: https://phabricator.wikimedia.org/T405892

Abban reviewed (changes requested):
The popup looks good, you're nearly there. Can you make the dark mode form use the same colours as the [EN Desktop banner](https://en.wikipedia.org/wiki/Main_Page?banner=B24_WMDE_Desktop_EN_05_ctrl&devMode&vectornightmode=1)? They look a bit weird being lighter like that and I know we tested the contrasts on the EN one last year.

Abban commented:
Like this:

<img width="573" height="517" alt="Screenshot 2025-10-21 at 07 46 56" src="https://github.com/user-attachments/assets/bdd21cd6-c79f-4eb2-bc0f-65387977a4e9" />

- Label background: `#292929`
- Label border: `#383838`
- Label background checked: `#20303f`
- LAbel border checked: 2px `#436fc7`

Sperling-0 committed:
Add size issue check

- This makes banners not appear on small screen

Ticket: https://phabricator.wikimedia.org/T405892

moiikana commented:
the unit tests should include, in case the link clicking should be tracked here or not
```
describe( 'Already Donated', () => {
		test.each( [
			[ 'expectFiresMaybeLaterEventOnLinkClick' ]
		] )( '%s', async ( testName: string ) => {
			await alreadyDonatedLinkFeatures[ testName ]( getWrapper() );
		} );
	} );
```

Sperling-0 committed:
Use correct 'feature' for already donated event

Sperling-0 committed:
Use correct 'feature' for already donated event for previous banners

- Update content

Ticket: https://phabricator.wikimedia.org/T405916

Sperling-0 commented:
> [!NOTE]  
> This banner uses correct 'feature' for already donated event.

Sperling-0 committed:
Already donated link only emits one event in the browser

- Update test accordingly

Ticket: https://phabricator.wikimedia.org/T405892#:~:text=Both%20banners%20send%20two%20events

Abban commented on src/themes/Treedip/DonationForm/Forms/MainDonationFormAmountButtonHighlighted.scss:
```diff
@@ -0,0 +1,63 @@
+	&-preferred-sticky {
+		position: absolute;
+		top: -27px;
+		left: 55%;
```
```suggestion
		right: 0;
```

Sperling-0 committed:
Fix sticky-note

- On real devices the sticky was half hidden
- Update content

Ticket: https://phabricator.wikimedia.org/T405892
